internal/controller/controllers/error_utils.go

```diff
@@ -30,3 +32,12 @@ func isClientError(err error) bool {
 		return false
 	}
 }
+
+func IsHTTPError(err error, httpErrorCode int) bool {
+	switch serr := err.(type) {
+	case *common.ApiErrorResponse:
```

Comment:
maybe you should add another case with `common.InfraErrorResponse`
Just to cover all the cases, currently it's relevant for 401/403 but because this function is generic it should cover both.

Reply:
Done